### PR TITLE
Add the Spotless plugin to support automatically removing unused imports

### PR DESCRIPTION
**Pull request by JLLeitschuh**
### Context

Often when contributing to Gradle/Gradle, I am annoyed that, after trying a bunch of things, I end up with a bunch of lingering imports. This causes checkstyle to fail. This significantly simplifies the fixing of these problems by allowing you to simply invoke `spotlessApply`.

### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [ ] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [ ] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -353,6 +354,42 @@
             <sha256 value="be46341a814e786c14c5d0e6cf5d470e55fff468b99509437c4014d8b2a20ce4" origin="Generated by Gradle because artifact wasn't signed"/>
          </artifact>
       </component>
+      <component group="com.diffplug.durian" name="durian-collect" version="1.2.0">
+         <artifact name="durian-collect-1.2.0.jar">
+            <pgp value="8756c4f765c9ac3cb6b85d62379ce192d401ab61"/>
```
Comment (melix) [inline gradle/verification-metadata.xml:359]: You added `^com[.]diffplug($|([.].*))` to the `8756c4f765c9ac3cb6b85d62379ce192d401ab61` trusted key, so adding the `com.diffplugin` entries here shouldn't be necessary. Does it fail without?

Comment (JLLeitschuh) [reply gradle/verification-metadata.xml:359]: I just ran the generation task, if it's not necessary, why does the generate task add them? Sounds like a bug?

Comment (melix) [reply gradle/verification-metadata.xml:359]: There are cases where the automated generation is not able to reduce properly.

Comment (JLLeitschuh) [reply gradle/verification-metadata.xml]: Fixed, and everything seems to work fine

**Comment by JLLeitschuh** [comment]
@bot-gradle help

**Comment by JLLeitschuh** [comment]
@bot-gradle test SanityCheck plz

**Comment by JLLeitschuh** [comment]
@bot-gradle test this please

**Review by melix — APPROVED** [review]

**Comment by JLLeitschuh** [comment]
@gradle-bot merge

**Pull request merged**